framework/validators/UniqueValidator.php

```diff
@@ -133,7 +145,24 @@ public function validateAttribute($model, $attribute)
         }
 
         if ($exists) {
-            $this->addError($model, $attribute, $this->message);
+            if (is_array($targetAttribute)) {
+                $attributeCombo = [];
+                $valueCombo = [];
+                foreach ($targetAttribute as $k => $v) {
+                    if(is_int($k)) {
+                        $attributeCombo[] = $v;
+                        $valueCombo[] = $model->$v;
+                    } else {
+                        $attributeCombo[] = $k;
+                        $valueCombo[] = $model->$k;
+                    }
+                }
+                array_walk($attributeCombo, function (&$attribute) use($model) { $attribute = $model->getAttributeLabel($attribute); });
+                array_walk($valueCombo, function (&$value) { $value = '"' . $value . '"'; });
```

Comment:
The reason I coded it outside a loop was to prevent duplication of quoting char (") for value but for performance reasons I can agree that it may be better to avoid `array_walk` and closure and put it inside a loop.